### PR TITLE
Don't solo selected staves during audio export

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #14123
Resolves: #14265
It's a somewhat dirty fix, but I'm not sure if we can do much better right now.

### CODE REVIEW

**Comment by vpereverzev** [comment]
1. Remove dependency from IPlaybackController as well as setExportingAudio method
2. Add notification to IPlayback facade `virtual async::Channel<RenderMode> modeChanged() const = 0;`
3. Add implementation of this method to playback.h
4. Subscribe Playback facade 
```
AudioEngine::instance()->modeChanged().onNotify([this]() {
    m_modeChangedChannel.send(AudioEngine::instance()->mode());
}
```
5. Subscribe PlaybackController on IPlayback::modeChanged notifications and do your things.

AudioEngine is the only one who knows what is the current mode and why, also it's not desirable to add dependencies between modules on IPlaybackController unless we don't have any other options

**Comment by cbjeukendrup** [comment]
@vpereverzev I'm afraid of one problem: will it not be too late if PlaybackController listens for notifications from the audio engine?
Wouldn't we get something like this?

1. [audio worker thread] set render mode to offline
2. [audio worker thread] immediately start rendering audio file
3. [main thread] oh, uhm, hey, the render mode has been changed. Let's update the mute states of the tracks. 
Oh, oops, audio export was already started, so now it's wrong for the first few milliseconds of the audio file.

Or is that nonsense?

**Comment by MarcSabatella** [comment]
FWIW, at least in the past, a number of other commands that had issues with active selection would simply clear the selection before executing, then restore it afterwards.  So, that's another approach to try if this one seems not right.

**Comment by cbjeukendrup** [comment]
@vpereverzev I just made this a little bit worse so that it now also resolves #14265 😄

File: src/playback/internal/playbackcontroller.h [hunk 1]
```diff
@@ -211,6 +213,9 @@ class PlaybackController : public IPlaybackController, public actions::Actionabl
 
     framework::Progress m_loadingProgress;
     std::list<engraving::InstrumentTrackId> m_loadingTracks;
+
+    /// The notation for which we're currently exporting audio
+    notation::INotationPtr m_exportedNotation = nullptr;
```
Comment (RomanPudashkin) [inline src/playback/internal/playbackcontroller.h:218]: You should use INotationWeakPtr here

**Comment by vpereverzev** [comment]
@cbjeukendrup 
I guess it's fine to work this out via playback controller in the context of 2 attached issues. The only problem I can see here - instead of `setExportedNotation ` method it'd be better to switch "current notation" from global context onto desired one. PlaybackController would react on thins notification at "PlaybackController::onNotationChanged()". Then when exporting has been finished - switch current notation back

**Comment by vpereverzev** [comment]
> @cbjeukendrup I guess it's fine to work this out via playback controller in the context of 2 attached issues. The only problem I can see here - instead of `setExportedNotation ` method it'd be better to switch "current notation" from global context onto desired one. PlaybackController would react on thins notification at "PlaybackController::onNotationChanged()". Then when exporting has been finished - switch current notation back

have you tried that already?

**Comment by vpereverzev** [comment]
> > @cbjeukendrup I guess it's fine to work this out via playback controller in the context of 2 attached issues. The only problem I can see here - instead of `setExportedNotation ` method it'd be better to switch "current notation" from global context onto desired one. PlaybackController would react on thins notification at "PlaybackController::onNotationChanged()". Then when exporting has been finished - switch current notation back
> 
> have you tried that already?

if that causes some undesirable problems - add method "PlaybackController::setNotation" and call it from "PlaybackController::onNotationChanged" and from "AbstractAudioWriter::doWriteAndWait"

**Comment by cbjeukendrup** [comment]
@vpereverzev I think something like your second suggestion would be better indeed, because otherwise the whole UI would switch to the different notation, which is maybe not desirable and of course comes with a performance cost.

However, I still see some difficulties. In PlaybackController::onNotationChanged, the controller subscribes itself to a bunch of notifications, and I don't know if I should move this to the new `setNotation` method or not. 
- if I do move this code, then the controller will be subscribed to a lot of notifications for the exported notation, while that is not necessary at all.
- if I don't move this code, then the controller will still be subscribed to the notifications of the old notation, but `m_notation` is the exported notation. So actually the controller is in an inconsistent state in this case.

Since the UI is currently blocked during audio export, these notifications are probably not received anyway, but still it feels not like a very clean solution to me. At least it will not be more elegant than with a simple `setExportedNotation` method, I'm afraid. I'll give it another try though.

**Comment by cbjeukendrup** [comment]
@vpereverzev I gave it a try; please have a look :)

**Comment by DmitryArefiev** [comment]
Tested #14123 #14265 on Win10, Mac12, LinuxUbuntu 22.04 - **FIXED**

**Comment by vpereverzev** [comment]
I'd not cherry pick that one. @cbjeukendrup please make a separate PR for RC

**Pull request merged**